src/components/application_manager/src/commands/mobile/register_app_interface_request.cc

```diff
@@ -805,9 +812,21 @@ void RegisterAppInterfaceRequest::SendRegisterAppInterfaceResponseToMobile() {
     resumption = resumer.IsApplicationSaved(application->policy_app_id(),
                                             application->mac_address());
   }
+  AppHmiTypes hmi_types;
+  if ((*message_)[strings::msg_params].keyExists(strings::app_hmi_type)) {
+    smart_objects::SmartArray* hmi_types_ptr =
```

Comment:
@tkireva , I meant something like this :
`DCHECK_OR_RETURN_VOID( (*message_)[strings::msg_params][strings:app_hmi_type].getType() ==
            smart_objects::SmartType_Array);
smart_objects::SmartArray* hmi_types_ptr =
          (*message_)[strings::msg_params][strings::app_hmi_type].asArray();`